app/src/main/java/ch/epfl/qedit/view/answer/MatrixFragment.java

```diff
@@ -0,0 +1,107 @@
+package ch.epfl.qedit.view.answer;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.text.InputFilter;
+import android.text.InputType;
+import android.text.method.DigitsKeyListener;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+import android.widget.EditText;
+import android.widget.TableLayout;
+import android.widget.TableRow;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import ch.epfl.qedit.R;
+import ch.epfl.qedit.model.MatrixFormat;
+import java.util.ArrayList;
+
+public class MatrixFragment extends Fragment {
+    private TableLayout tableLayout;
+    private boolean hasDecimal = true;
```

Comment:
Do we need `hasDecimal`, `hasSign`, `tableRowNumbers`, ... as private fields, since `matrixFormat` already contains that information ?

Reply:
I thought about that and I added those because I wanted the code to be clearer and cleaner than having in some places

`editText.setKeyListener(DigitsKeyListener.getInstance(matrixFormat.hasSign, matrixFormat.hasDecimal));
`
I think here using only `hasSign` and `hasDecimal` is cleaner

`editText.setFilters(new InputFilter[] {new InputFilter.LengthFilter(matrixFormat.maxCharacters)});`

Reply:
I agree with you, writing `getXX` instead of `XX` is annoying. But having those fields in the fragment is a waste of space. It also means that extra code needs to be run to "import" those attributes, when you call `setValues`.
IMO the cons outweigh the pros, but that's not such a big issue in the end.